docs/gallery/steane_encoding.ipynb

```diff
@@ -883,22 +179,103 @@
     "\n",
-    "![Z0](../media/gallery/steane_encoding/steane_encoding_result_Z_observable_0.png)\n",
+    "def generate_graphs(support_observable_basis: Basis) -> None:\n",
+    "    block_graph = steane_encoding(support_observable_basis)\n",
+    "    zx_graph = block_graph.to_zx_graph()\n",
     "\n",
-    "![Z1](../media/gallery/steane_encoding/steane_encoding_result_Z_observable_1.png)\n",
+    "    correlation_surfaces = block_graph.find_correlation_surfaces()\n",
     "\n",
-    "![Z2](../media/gallery/steane_encoding/steane_encoding_result_Z_observable_2.png)\n",
+    "    # Start simulation using sinter, note that the `save_resume_filepath` is useful\n",
+    "    # when simulation takes a long time.\n",
+    "    # If the Python interpreter is stopped or killed, calling this method again with\n",
+    "    # the same file path will load the previous results and resume from where it left off.\n",
+    "    stats = start_simulation_using_sinter(\n",
+    "        block_graph,\n",
+    "        range(1, 4),\n",
+    "        list(numpy.logspace(-4, -1, 10)),\n",
+    "        NoiseModel.uniform_depolarizing,\n",
+    "        manhattan_radius=2,\n",
+    "        observables=correlation_surfaces,\n",
+    "        num_workers=20,\n",
+    "        max_shots=10_000_000,\n",
+    "        max_errors=5_000,\n",
+    "        decoders=[\"pymatching\"],\n",
+    "        save_resume_filepath=Path(\"../_examples_database/steane_stats\"),\n",
+    "        database_path=Path(\"../_examples_database/steane_database\"),\n",
+    "    )\n",
     "\n",
-    "![Z3](../media/gallery/steane_encoding/steane_encoding_result_Z_observable_3.png)\n"
+    "    for i, stat in enumerate(stats):\n",
+    "        fig, ax = plt.subplots()\n",
+    "        sinter.plot_error_rate(\n",
+    "            ax=ax,\n",
+    "            stats=stat,\n",
+    "            x_func=lambda stat: stat.json_metadata[\"p\"],\n",
+    "            failure_units_per_shot_func=lambda stat: stat.json_metadata[\"d\"],\n",
+    "            group_func=lambda stat: stat.json_metadata[\"d\"],\n",
+    "        )\n",
+    "        plot_observable_as_inset(ax, zx_graph, correlation_surfaces[i])\n",
+    "        ax.grid(axis=\"both\")\n",
+    "        ax.legend()\n",
+    "        ax.loglog()\n",
+    "        ax.set_title(\"Steane Encoding Error Rate\")\n",
+    "        ax.set_xlabel(\"Physical Error Rate\")\n",
+    "        ax.set_ylabel(\"Logical Error Rate(per round)\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Z Basis"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "generate_graphs(Basis.Z)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### X Basis"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "generate_graphs(Basis.X)"
```

Comment:
This code block led to a build error. https://tqec.github.io/tqec/pull/592/gallery/steane_encoding.html#X-Basis